http-clients/src/main/java/com/palantir/remoting3/clients/ClientConfiguration.java

```diff
@@ -70,6 +82,13 @@
      */
     Duration backoffSlotSize();
 
+    @Value.Check
+    default void check() {
+        if (meshProxy().isPresent()) {
+            checkArgument(maxNumRetries() == 0, "If meshProxy is configured then maxNumRetries must be 0");
```

Comment:
I think we have some constraints regarding the target URI?

Reply:
added + added tests